src/backend/distributed/deparser/deparse.c

```diff
@@ -0,0 +1,115 @@
+
+#include "postgres.h"
+
+#include "distributed/deparser.h"
+
+static const char * deparse_drop_stmt(DropStmt *stmt);
+static const char * deparse_alter_table_stmt(AlterTableStmt *stmt);
+static const char * deparse_rename_stmt(RenameStmt *stmt);
+
+
+/*
+ * DeparseTreeNode aims to be the inverse of postgres' ParseTreeNode. Currently with
+ * limited support. Check support before using, and add support for new statements as
+ * required.
+ *
+ * Currently supported:
+ *  - CREATE TYPE
+ *  - ALTER TYPE
+ *  - DROP TYPE
+ */
+const char *
+DeparseTreeNode(Node *stmt)
```

Comment:
At some point we will likely want to pass a deparsing context (e.g. to store the shard ID to use when deparsing table names). Not on the critical path, but it might make sense to wire that in early.

Reply:
Discussed offline, we can probably do these translations in the tree because utility commands have raw strings rather than OIDs.